torchtext/experimental/datasets/raw/common.py

```diff
@@ -0,0 +1,44 @@
+import torch
+
+
+class RawTextIterableDataset(torch.utils.data.IterableDataset):
+    """Defines an abstraction for raw text iterable datasets.
+    """
+
+    def __init__(self, name, full_num_lines, iterator):
+        """Initiate text-classification dataset.
+        """
+        super(RawTextIterableDataset, self).__init__()
+        self.name = name
+        self.full_num_lines = full_num_lines
+        self._iterator = iterator
+        self.has_setup = False
+        self.start = 0
+        self.num_lines = None
+
+    def setup_iter(self, start=0, num_lines=None):
```

Comment:
We could also add [DataLoader support](https://github.com/pytorch/text/blob/master/examples/text_classification/iterable_train.py#L184).